File: 2025-05-06/Find_The_Town_Judge.java
public class Find_The_Town_Judge {
    public static int findJudge(int n, int[][] trust) {
        int[] peopleTrust = new int[n+1];

        for(int[] t : trust) {
            peopleTrust[t[0]]--;
            peopleTrust[t[1]]++;
        }

        for(int i=0;i<=n;i++)
            if(peopleTrust[i] == n-1)
                return i;

        return -1;
    }
    public static void main(String[] args) {
        int[][] people = {{1,3},{2,3}};
        System.out.println(findJudge(3, people));
    }
}

File: 2026-03-05/Find_The_Town_Judge.py
from typing import List


class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        is_trusting = [False] * (n+1)
        trust_freq = [0] * (n+1)

        for a, b in trust:
            is_trusting[a] = True
            trust_freq[b] += 1

        for i in range(n+1):
            if not is_trusting[i] and trust_freq[i] == n - 1:
                return i

        return -1


sol = Solution()
print(sol.findJudge(3, [[1,3],[2,3]]))